sound/soc/sof/sof-acpi-dev.c

```diff
@@ -93,21 +93,16 @@ int sof_acpi_probe(struct platform_device *pdev, const struct sof_dev_desc *desc
 		sof_pdata->tplg_filename_prefix =
 			sof_pdata->desc->default_tplg_path;
 
-#if IS_ENABLED(CONFIG_SND_SOC_SOF_PROBE_WORK_QUEUE)
-	/* set callback to enable runtime_pm */
+	/* set callback to be called on successful device probe to enable runtime_pm */
 	sof_pdata->sof_probe_complete = sof_acpi_probe_complete;
-#endif
+
 	/* call sof helper for DSP hardware probe */
 	ret = snd_sof_device_probe(dev, sof_pdata);
 	if (ret) {
 		dev_err(dev, "error: failed to probe DSP hardware!\n");
 		return ret;
 	}
 
-#if !IS_ENABLED(CONFIG_SND_SOC_SOF_PROBE_WORK_QUEUE)
-	sof_acpi_probe_complete(dev);
-#endif
-
 	return ret;
```

Comment:
Either:
`
	return snd_sof_device_probe(dev, sof_pdata);
`

or if the print is needed:
```
	ret = snd_sof_device_probe(dev, sof_pdata);
	if (ret)
		dev_err(dev, "error: failed to probe DSP hardware!\n");

	return ret;
}
```

Reply:
I prefer the version with the print. it's awful when the probe fails silently.

Reply:
but all 3 (ACPI, PCI, OF) versions have the same code calling probe, checking return code, printing the same error message, so, maybe move that error message into `snd_sof_device_probe()`? Can be a separate patch though